### PR TITLE
If you're gonna fail, might as well fail well.  Closes #891

### LINKED ISSUE #891: JES: Files in output expressions not found
The following workflow fails:

This workflow fails:
```
task MakeMeAFile {
    command <<<
      echo FILECONTENT > output.txt
      >>>

    runtime {
      docker: "ubuntu:latest"
      memory: "1 GB"
      preemptible: 3
    }

    output {
        String out = read_string("output.txt")
    }
}

task ReadMeAFile {
    File infile

    command <<<
        cat ${infile}
    >>>

    runtime {
      docker: "ubuntu:latest"
      memory: "1 GB"
      preemptible: 3
    }

    output {
        File out = read_string(stdout())
    }
}

workflow FileMakeAndRead {
    call MakeMeAFile
    call ReadMeAFile { input: infile = MakeMeAFile.out }
}
```

Because:
```
"failures": [
  "Could not find suitable filesystem to parse output.txt"
]
```

### PR DESCRIPTION
**Pull request by scottfrazer**

### CODE REVIEW

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/io/package.scala [hunk 1]
```diff
@@ -29,7 +29,7 @@ package object io {
   }
 
   private [jes] def isFatalJesException(t: Throwable): Boolean = t match {
-    case e: HttpResponseException if e.getStatusCode == 403 => true
+    case e: HttpResponseException if e.getStatusCode == 403 || e.getStatusCode == 400 => true
```
Comment (mcovarr) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/io/package.scala]: is it safe to generally treat 400 as a fatal JES exception?  or perhaps only for localization / delocalization stuff?

Comment (scottfrazer) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/io/package.scala]: I don't know

**Comment by cjllanwarne** [comment]
👍

**Comment by mcovarr** [comment]
:+1:

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/io/package.scala (hunk 1 above)
Comment (scottfrazer) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/io/package.scala]: Yeah, so it seems this is the response we get back:

```json
{
  "code" : 400,
  "errors" : [ {
    "domain" : "global",
    "message" : "Pipeline 7425209317495770049: Unable to evaluate parameters: %!(EXTRA string=parameter \"infile-0\" has invalid value: FILECONTENT)",
    "reason" : "badRequest"
  } ],
  "message" : "Pipeline 7425209317495770049: Unable to evaluate parameters: %!(EXTRA string=parameter \"infile-0\" has invalid value: FILECONTENT)",
  "status" : "INVALID_ARGUMENT"
}
```

I guess I can look for `INVALID_ARGUMENT`

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/io/package.scala [hunk 2]
```diff
@@ -30,6 +30,7 @@ package object io {
 
   private [jes] def isFatalJesException(t: Throwable): Boolean = t match {
     case e: HttpResponseException if e.getStatusCode == 403 => true
+    case e: HttpResponseException if e.getStatusCode == 400 && e.getContent.contains("INVALID_ARGUMENT") => true
```
Comment (scottfrazer) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/io/package.scala]: @mcovarr -- This is the best I could do, but it feels ugly.  I was surprised that I couldn't find this as a string on its own

**Comment by ruchim** [comment]
Since Scott is out and this branch appears ready, it's getting merged so it doesn't linger for 2 weeks.

**Pull request merged**